doc/user/posix.md

```diff
@@ -187,10 +185,36 @@ Enabling this option will turn on the following features:
 * Readahead in dfuse and inserting data into kernel cache
 * MMAP write optimization
 
-To turn on caching use the `--enable-caching` command-line option for dfuse. This
-will enable the feature for all accessed containers.  When this option is used,
-the containers accessed should only be accessed from one node, so it may
-be necessary to create a container per node in this model.
+To selectively control caching within a container the following container
+attributes should be used, if any attribute is set then the rest are assumed
+to be set to 0 or off, except dentry-dir-time which defaults to dentry-time
+
+| **Attribute name**    | **Description**                                                  |
+| --------------------- | ---------------------------------------------------------------- |
+| dfuse-attr-time       | How long file attributes are cached                              |
+| dfuse-dentry-time     | How long directory entries are cached                            |
+| dfuse-dentry-dir-time | How long dentries are cached, if the entry is itself a directory |
+| dfuse-ndentry-time    | How long negative dentries are cached                            |
+| dfuse-data-cache      | Data caching enabled for this file ("on"/"off")                  |
+
+For metadata caching attributes specify the duration that the cache should be
+valid for, specified in seconds, and allowing 'S' or 'M' suffix.
+
+dfuse-data-cache should be set to "on", or "off" if set, any other value will
+log an error, and result in the cache being off.
+
+With no options specified attr and dentry timeouts will be 1 second, dentry-dir
+and ndentry timeouts will be 5 seconds, and data caching will be enabled.
+
+These are two command line options to control the DFuse process itself.
+
+| **Command line option | **Description**           |
+| --------------------- | ------------------------- |
+| --disable-caching     | Disables all caching      |
+| --disable-wb-caching  | Disables write-back cache |
+
+These will affect all containers accessed via DFuse, regardless of any
+container attributes.
```

Comment:
I would add a:

!note
     Caching is enabled by default in dfuse. This might cause some parallel applications to fail. Please use disable caching ...

Reply:
Please see below an example of what others are saying in their documentation:
https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/docs/semantics.md#caching

Reply:
I'm doing another pass on this, and will add what we discovered on aio at the same time.